File: pocketr-java.api/src/test/java/com/decrux/pocketr/api/entities/db/ledger/LedgerTxnTest.java
package com.decrux.pocketr.api.entities.db.ledger;

import java.time.Instant;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

class LedgerTxnTest {

    @Test
    void onCreateRefreshesUpdatedAt() {
        Instant originalUpdatedAt = Instant.parse("2020-01-01T00:00:00Z");
        LedgerTxn txn = new LedgerTxn();
        txn.setUpdatedAt(originalUpdatedAt);

        txn.onCreate();

        assertTrue(txn.getUpdatedAt().isAfter(originalUpdatedAt));
    }

    @Test
    void onUpdateRefreshesUpdatedAt() {
        Instant originalUpdatedAt = Instant.parse("2020-01-01T00:00:00Z");
        LedgerTxn txn = new LedgerTxn();
        txn.setUpdatedAt(originalUpdatedAt);

        txn.onUpdate();

        assertTrue(txn.getUpdatedAt().isAfter(originalUpdatedAt));
    }
}

File: pocketr-api/src/test/kotlin/com/decrux/pocketr/api/entities/db/ledger/LedgerTxnTest.kt
package com.decrux.pocketr.api.entities.db.ledger

import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test
import java.time.Instant

class LedgerTxnTest {
    @Test
    fun onCreateRefreshesUpdatedAt() {
        val originalUpdatedAt = Instant.parse("2020-01-01T00:00:00Z")
        val txn = LedgerTxn(updatedAt = originalUpdatedAt)

        txn.onCreate()

        assertTrue(txn.updatedAt.isAfter(originalUpdatedAt))
    }

    @Test
    fun onUpdateRefreshesUpdatedAt() {
        val originalUpdatedAt = Instant.parse("2020-01-01T00:00:00Z")
        val txn = LedgerTxn(updatedAt = originalUpdatedAt)

        txn.onUpdate()

        assertTrue(txn.updatedAt.isAfter(originalUpdatedAt))
    }
}
